controllers/storagecluster/cephcluster.go

```diff
@@ -605,13 +605,11 @@ func getNetworkSpec(sc ocsv1.StorageCluster) rookCephv1.NetworkSpec {
 	// respect both the old way and the new way for enabling HostNetwork
 	networkSpec.HostNetwork = networkSpec.HostNetwork || sc.Spec.HostNetwork
 
-	// If it's not an external and not a provider cluster always require msgr2
-	if !sc.Spec.AllowRemoteStorageConsumers && !sc.Spec.ExternalStorage.Enable {
-		if networkSpec.Connections == nil {
-			networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
-		}
-		networkSpec.Connections.RequireMsgr2 = true
+	if networkSpec.Connections == nil {
+		networkSpec.Connections = &rookCephv1.ConnectionsSpec{}
 	}
+	// Always require msgr2 for all modes of ODF
+	networkSpec.Connections.RequireMsgr2 = true
```

Comment:
I am not sure we can blindly change that for provider mode. What happens to existing deployments?

Reply:
For existing deployments the mons after the upgrade would have the v1 port still available, so there would be no problem for the existing volumes that are already mounted. For any new volumes that are created after the upgrade, the volume would use the v2 port.

Reply:
@malayparida2000  I am not sure that will work out of the box, don't we need to inform the CSI running on the clients that we moved to v2? If so how do we do that in provider mode? the CSI is running on many different clusters

Reply:
> would now get the new info from the provider cluster
- this config will be pushed to clients right after provider gets upgraded but not when the client gets upgraded
- nevertheless, based on all the info presented even 4.17 clients will have v2 endpoint which should be supported by kernel on client clusters

> don't we need to inform the CSI running on the clients that we moved to v2
- two changes will happen after provider upgrade, new v2 endpoints and mount option `ms_mode=prefer-crc` will be pushed to client clusters and IIUC they can be changed on the fly w/o any issues for both existing and new workloads.